core/src/main/java/io/seata/core/protocol/MergedWarpMessage.java

```diff
@@ -49,21 +51,21 @@ public short getTypeCode() {
 
     @Override
     public byte[] encode() {
-        int bufferSize = msgs.size() * 1024;
-        ByteBuffer byteBuffer = ByteBuffer.allocate(bufferSize);
-        byteBuffer.putShort((short)msgs.size());
-        for (AbstractMessage msg : msgs) {
-            //msg.setChannelHandlerContext(ctx);
-            byte[] data = msg.encode();
-            byteBuffer.putShort(msg.getTypeCode());
-            byteBuffer.put(data);
+        final ByteBuf buffer = Unpooled.buffer();
```

Comment:
default buffer size=256，can we set a larger size and reduce the number of resize?

Reply:
@slievrly maybe set to what we hardcoded before? (say 1024)